### PR TITLE
13085 add more logging to ess fulfilment process

### PR DESCRIPTION
**Pull request by Siddartha-tech**
Added logs while Querying and Download ENC files from File Share Service

### CODE REVIEW

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/Services/FulfilmentFileShareService.cs [hunk 1]
```diff
@@ -76,8 +80,10 @@ public async Task DownloadFileShareServiceFiles(SalesCatalogueServiceResponseQue
...
                 var downloadPath = Path.Combine(exchangeSetRootPath, item.ProductName.Substring(0, 2), item.ProductName, Convert.ToString(item.EditionNumber), Convert.ToString(item.UpdateNumber));
                 await fileShareService.DownloadBatchFiles(item.FileUri, downloadPath, message);
+                logger.LogInformation(EventIds.FileShareServicePreparingToDownloadENCFilesCompleted.ToEventId(), "File share service request completed preparing and downloading ENC files for BatchId:{batchId} based on Product/CellName:{ProductName}, EditionNumber:{EditionNumber} and UpdateNumber:{UpdateNumber} with \n Href: [{FileUri}] and _X-Correlation-ID:{CorrelationId}", message.BatchId, item.ProductName, item.EditionNumber, item.UpdateNumber, item.FileUri, message.CorrelationId);
```
Comment (barrie-cooper) [inline UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/Services/FulfilmentFileShareService.cs:86]: Please change text to something like:
>Completed file share service request for BatchId:{batchId} based on Product/CellName:{ProductName}, EditionNumber:{EditionNumber} and UpdateNumber:{UpdateNumber} with \n Href: [{FileUri}] and _X-Correlation-ID:{CorrelationId}

**Review by barrie-cooper — COMMENTED** [review]
Some textual changes needed I think.

Also please shorten the lines making the `logger.LogInformation(...)` calls as they are too long (~150 characters).

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs [hunk 2]
```diff
@@ -109,14 +109,15 @@ public async Task<SearchBatchResponse> GetBatchInfoBasedOnProducts(List<Products
...
-            var uri = $"/batch?limit={fileShareServiceConfig.Value.Limit}&start={fileShareServiceConfig.Value.Start}&$filter={fileShareServiceConfig.Value.ProductCode} {productWithAttributes}";
+            var uri = $"/batch?limit={fileShareServiceConfig.Value.Limit}&start={fileShareServiceConfig.Value.Start}&$filter={fileShareServiceConfig.Value.ProductCode} {productWithAttributes.Item1}";
 
             HttpResponseMessage httpResponse;
 
             string payloadJson = string.Empty;
             var productList = new List<string>();
             var prodCount = products.Select(a => a.UpdateNumbers).Sum(a => a.Count);
             int queryCount = 0;
+            logger.LogInformation(EventIds.FileShareServicePreparingToQueryENCFilesStart.ToEventId(), "File share service request started preparing to querying ENC files for BatchId:{batchId} based on {sbLog} and _X-Correlation-ID:{CorrelationId}", batchId, productWithAttributes.Item2, correlationId);
```
[the remark this answers is not in the document]
Comment (Siddartha-tech) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs:120]: Yes. Did changes accordingly.

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Logging/EventIds.cs [hunk 3]
```diff
@@ -496,7 +496,23 @@ public enum EventIds
         /// <summary>
         /// 805122 - New access token is added to cache for external end point resource.
         /// </summary>
-        CachingExternalEndPointToken = 805122
+        CachingExternalEndPointToken = 805122,
+        /// <summary>
+        ///  805123 - Start preparing to downloading ENC files based on Product/CellName EditionNumber and UpdateNumber from file share service.
```
[the remark this answers is not in the document]
Comment (Siddartha-tech) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Logging/EventIds.cs]: Yes. Did changes accordingly. Thanks!

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Logging/EventIds.cs [hunk 4]
```diff
@@ -496,7 +496,23 @@ public enum EventIds
...
+        /// </summary>
+        FileShareServicePreparingToDownloadENCFilesStart = 805123,
+        /// <summary>
+        ///  805124 - Completed preparing and downloading ENC files based on Product/CellName EditionNumber and UpdateNumber from file share service.
+        /// </summary>
+        FileShareServicePreparingToDownloadENCFilesCompleted = 805124,
+        /// <summary>
+        /// 805125 - Start preparing to Quering ENC files based on Product/CellName EditionNumber and UpdateNumber from file share service.
```
[the remark this answers is not in the document]
Comment (Siddartha-tech) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Logging/EventIds.cs]: Yes. Did changes accordingly. Thanks!

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/Services/FulfilmentFileShareService.cs [hunk 5]
```diff
@@ -76,8 +80,10 @@ public async Task DownloadFileShareServiceFiles(SalesCatalogueServiceResponseQue
         {
             foreach (var item in fulfilmentDataResponse)
             {
+                logger.LogInformation(EventIds.FileShareServicePreparingToDownloadENCFilesStart.ToEventId(), "File share service request started preparing to download ENC files for BatchId:{batchId} based on Product/CellName:{ProductName}, EditionNumber:{EditionNumber} and UpdateNumber:{UpdateNumber} with \n Href: [{FileUri}] and _X-Correlation-ID:{CorrelationId}", message.BatchId, item.ProductName, item.EditionNumber, item.UpdateNumber, item.FileUri, message.CorrelationId);
```
[the remark this answers is not in the document]
Comment (Siddartha-tech) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/Services/FulfilmentFileShareService.cs]: Yes. Did changes accordingly. Thanks!

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs [hunk 6]
```diff
@@ -134,6 +135,7 @@ public async Task<SearchBatchResponse> GetBatchInfoBasedOnProducts(List<Products
             } while (httpResponse.IsSuccessStatusCode && internalSearchBatchResponse.Entries.Count != 0 && internalSearchBatchResponse.Entries.Count < prodCount && !string.IsNullOrWhiteSpace(uri));
             internalSearchBatchResponse.QueryCount = queryCount;
             CheckProductsExistsInFileShareService(products, correlationId, batchId, internalSearchBatchResponse, internalNotFoundProducts, prodCount);
+            logger.LogInformation(EventIds.FileShareServicePreparingToQueryENCFilesCompleted.ToEventId(), "File share service request completed preparing and querying ENC files for BatchId:{batchId} based on {sbLog} and _X-Correlation-ID:{CorrelationId}", batchId, productWithAttributes.Item2, correlationId);
```
[the remark this answers is not in the document]
Comment (Siddartha-tech) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs]: Yes. Did changes accordingly. Thanks!

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Logging/EventIds.cs [hunk 7]
```diff
@@ -496,7 +496,23 @@ public enum EventIds
...
         /// 805122 - New access token is added to cache for external end point resource.
         /// </summary>
-        CachingExternalEndPointToken = 805122
+        CachingExternalEndPointToken = 805122,
+        /// <summary>
+        ///  805123 - Start preparing to downloading ENC files based on Product/CellName EditionNumber and UpdateNumber from file share service.
+        /// </summary>
+        FileShareServicePreparingToDownloadENCFilesStart = 805123,
+        /// <summary>
+        ///  805124 - Completed preparing and downloading ENC files based on Product/CellName EditionNumber and UpdateNumber from file share service.
```
[the remark this answers is not in the document]
Comment (Siddartha-tech) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Logging/EventIds.cs]: Yes. Did changes accordingly. Thanks!

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Logging/EventIds.cs [hunk 8]
```diff
@@ -496,7 +496,23 @@ public enum EventIds
...
+        /// </summary>
+        FileShareServicePreparingToDownloadENCFilesCompleted = 805124,
+        /// <summary>
+        /// 805125 - Start preparing to Quering ENC files based on Product/CellName EditionNumber and UpdateNumber from file share service.
+        /// </summary>
+        FileShareServicePreparingToQueryENCFilesStart = 805125,
+        /// <summary>
+        /// 805126 - Completed preparing and Quering ENC files based on Product/CellName EditionNumber and UpdateNumber from file share service.
```
[the remark this answers is not in the document]
Comment (Siddartha-tech) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Logging/EventIds.cs]: Yes. Did changes accordingly. Thanks!

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/Services/FulfilmentFileShareService.cs (hunk 1 above)
Comment (Siddartha-tech) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/Services/FulfilmentFileShareService.cs]: Yes. Did changes accordingly. Thanks!

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs [hunk 9]
```diff
@@ -109,14 +109,15 @@ public async Task<SearchBatchResponse> GetBatchInfoBasedOnProducts(List<Products
...
-            var uri = $"/batch?limit={fileShareServiceConfig.Value.Limit}&start={fileShareServiceConfig.Value.Start}&$filter={fileShareServiceConfig.Value.ProductCode} {productWithAttributes}";
+            var uri = $"/batch?limit={fileShareServiceConfig.Value.Limit}&start={fileShareServiceConfig.Value.Start}&$filter={fileShareServiceConfig.Value.ProductCode} {productWithAttributes.Item1}";
 
             HttpResponseMessage httpResponse;
 
             string payloadJson = string.Empty;
             var productList = new List<string>();
             var prodCount = products.Select(a => a.UpdateNumbers).Sum(a => a.Count);
             int queryCount = 0;
+            logger.LogInformation(EventIds.FileShareServicePreparingToQueryENCFilesStart.ToEventId(), "Preparing file share service request for BatchId:{batchId} based on {sbLog} and _X-Correlation-ID:{CorrelationId}", batchId, productWithAttributes.Item2, correlationId);
```
Comment (vijendr13545) [inline UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs:120]: Log attribute `sbLog` is not much meaningful. Please can you revisit it.

Comment (Siddartha-tech) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs]: Yes updated.

**Review by barrie-cooper — APPROVED** [review]
Looks good to me

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/Services/FulfilmentFileShareService.cs [hunk 10]
```diff
@@ -76,8 +80,10 @@ public async Task DownloadFileShareServiceFiles(SalesCatalogueServiceResponseQue
         {
             foreach (var item in fulfilmentDataResponse)
             {
+                logger.LogInformation(EventIds.FileShareServicePreparingToDownloadENCFilesStart.ToEventId(), "Preparing file share service request for BatchId:{batchId} based on Product/CellName:{ProductName}, EditionNumber:{EditionNumber} and UpdateNumber:{UpdateNumber} with \n Href: [{FileUri}] and _X-Correlation-ID:{CorrelationId}", message.BatchId, item.ProductName, item.EditionNumber, item.UpdateNumber, item.FileUri, message.CorrelationId);
```
Comment (vijendr13545) [inline UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/Services/FulfilmentFileShareService.cs:83]:
```suggestion
                logger.LogInformation(EventIds.FileShareServicePreparingToDownloadENCFilesStart.ToEventId(), "Preparing file share service download request for BatchId:{batchId} based on Product/CellName:{ProductName}, EditionNumber:{EditionNumber} and UpdateNumber:{UpdateNumber} with \n Href: [{FileUri}] and _X-Correlation-ID:{CorrelationId}", message.BatchId, item.ProductName, item.EditionNumber, item.UpdateNumber, item.FileUri, message.CorrelationId);
```

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/Services/FulfilmentFileShareService.cs [hunk 11]
```diff
@@ -76,8 +80,10 @@ public async Task DownloadFileShareServiceFiles(SalesCatalogueServiceResponseQue
...
                 var downloadPath = Path.Combine(exchangeSetRootPath, item.ProductName.Substring(0, 2), item.ProductName, Convert.ToString(item.EditionNumber), Convert.ToString(item.UpdateNumber));
                 await fileShareService.DownloadBatchFiles(item.FileUri, downloadPath, message);
+                logger.LogInformation(EventIds.FileShareServicePreparingToDownloadENCFilesCompleted.ToEventId(), "Completed file share service request for BatchId:{batchId} based on Product/CellName:{ProductName}, EditionNumber:{EditionNumber} and UpdateNumber:{UpdateNumber} with \n Href: [{FileUri}] and _X-Correlation-ID:{CorrelationId}", message.BatchId, item.ProductName, item.EditionNumber, item.UpdateNumber, item.FileUri, message.CorrelationId);
```
Comment (vijendr13545) [inline UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/Services/FulfilmentFileShareService.cs:86]:
```suggestion
                logger.LogInformation(EventIds.FileShareServicePreparingToDownloadENCFilesCompleted.ToEventId(), "Completed file share service download request for BatchId:{batchId} based on Product/CellName:{ProductName}, EditionNumber:{EditionNumber} and UpdateNumber:{UpdateNumber} with \n Href: [{FileUri}] and _X-Correlation-ID:{CorrelationId}", message.BatchId, item.ProductName, item.EditionNumber, item.UpdateNumber, item.FileUri, message.CorrelationId);
```

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Logging/EventIds.cs [hunk 12]
```diff
@@ -496,7 +496,23 @@ public enum EventIds
...
-        CachingExternalEndPointToken = 805122
+        CachingExternalEndPointToken = 805122,
+        /// <summary>
+        ///  805123 -Preparing to download ENC files based on Product/CellName EditionNumber and UpdateNumber from file share service.
+        /// </summary>
+        FileShareServicePreparingToDownloadENCFilesStart = 805123,
+        /// <summary>
+        ///  805124 -Completed download of ENC files based on Product/CellName EditionNumber and UpdateNumber from file share service.
+        /// </summary>
+        FileShareServicePreparingToDownloadENCFilesCompleted = 805124,
```
Comment (vijendr13545) [inline UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Logging/EventIds.cs:507]: Is it for preparation or actual download complete?
```suggestion
        FileShareServiceDownloadENCFilesCompleted = 805124,
```

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Logging/EventIds.cs [hunk 13]
```diff
@@ -496,7 +496,23 @@ public enum EventIds
...
+        /// <summary>
+        /// 805125 - Preparing to query ENC files based on Product/CellName EditionNumber and UpdateNumber from file share service.
+        /// </summary>
+        FileShareServicePreparingToQueryENCFilesStart = 805125,
+        /// <summary>
+        /// 805126 - Completed query for ENC files based on Product/CellName EditionNumber and UpdateNumber from file share service.
+        /// </summary>
+        FileShareServicePreparingToQueryENCFilesCompleted = 805126
```
Comment (vijendr13545) [inline UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Logging/EventIds.cs:515]: Is it for preparation or actual query complete?

```suggestion
        FileShareServiceSearchQueryForSetOfENCsCompleted = 805126
```

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Logging/EventIds.cs [hunk 14]
```diff
@@ -496,7 +496,23 @@ public enum EventIds
...
+        /// </summary>
+        FileShareServicePreparingToDownloadENCFilesStart = 805123,
+        /// <summary>
+        ///  805124 -Completed download of ENC files based on Product/CellName EditionNumber and UpdateNumber from file share service.
+        /// </summary>
+        FileShareServicePreparingToDownloadENCFilesCompleted = 805124,
+        /// <summary>
+        /// 805125 - Preparing to query ENC files based on Product/CellName EditionNumber and UpdateNumber from file share service.
+        /// </summary>
+        FileShareServicePreparingToQueryENCFilesStart = 805125,
```
Comment (vijendr13545) [inline UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Logging/EventIds.cs:511]:
```suggestion
        FileShareServicePreparingToSearchSetOfENCsStarted = 805125,
```

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs [hunk 15]
```diff
@@ -109,14 +109,15 @@ public async Task<SearchBatchResponse> GetBatchInfoBasedOnProducts(List<Products
...
-            var uri = $"/batch?limit={fileShareServiceConfig.Value.Limit}&start={fileShareServiceConfig.Value.Start}&$filter={fileShareServiceConfig.Value.ProductCode} {productWithAttributes}";
+            var uri = $"/batch?limit={fileShareServiceConfig.Value.Limit}&start={fileShareServiceConfig.Value.Start}&$filter={fileShareServiceConfig.Value.ProductCode} {productWithAttributes.Item1}";
 
             HttpResponseMessage httpResponse;
 
             string payloadJson = string.Empty;
             var productList = new List<string>();
             var prodCount = products.Select(a => a.UpdateNumbers).Sum(a => a.Count);
             int queryCount = 0;
+            logger.LogInformation(EventIds.FileShareServicePreparingToQueryENCFilesStart.ToEventId(), "Preparing file share service request for BatchId:{batchId} based on {productDetails} and _X-Correlation-ID:{CorrelationId}", batchId, productWithAttributes.Item2, correlationId);
```
Comment (vijendr13545) [inline UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs:120]:
```suggestion
            logger.LogInformation(EventIds.FileShareServicePreparingToQueryENCFilesStart.ToEventId(), "Preparing file share service search request for {productDetails}. ESS BatchId:{batchId} and _X-Correlation-ID:{CorrelationId}",  productWithAttributes.Item2, batchId, correlationId);
```

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs [hunk 16]
```diff
@@ -134,6 +135,7 @@ public async Task<SearchBatchResponse> GetBatchInfoBasedOnProducts(List<Products
             } while (httpResponse.IsSuccessStatusCode && internalSearchBatchResponse.Entries.Count != 0 && internalSearchBatchResponse.Entries.Count < prodCount && !string.IsNullOrWhiteSpace(uri));
             internalSearchBatchResponse.QueryCount = queryCount;
             CheckProductsExistsInFileShareService(products, correlationId, batchId, internalSearchBatchResponse, internalNotFoundProducts, prodCount);
+            logger.LogInformation(EventIds.FileShareServicePreparingToQueryENCFilesCompleted.ToEventId(), "Completed file share service request for BatchId:{batchId} based on {productDetails} and _X-Correlation-ID:{CorrelationId}", batchId, productWithAttributes.Item2, correlationId);
```
Comment (vijendr13545) [inline UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs:138]: Please can we have log message something like:

```suggestion
            logger.LogInformation(EventIds.FileShareServicePreparingToQueryENCFilesCompleted.ToEventId(), "Completed file share service search request for {productDetails}. ESS BatchId:{batchId} and _X-Correlation-ID:{CorrelationId}",  productWithAttributes.Item2, batchId, correlationId);
```

**Review by vijendr13545 — COMMENTED** [review]
Please have at my review comments/suggestions. Looks good to me once they are resolved. Thanks.

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/Services/FulfilmentFileShareService.cs (hunk 10 above)
Comment (Siddartha-tech) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/Services/FulfilmentFileShareService.cs]: Yes updated message as per suggested, Thanks!

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/Services/FulfilmentFileShareService.cs (hunk 11 above)
Comment (Siddartha-tech) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/Services/FulfilmentFileShareService.cs]: Yes updated message as per suggested, Thanks!

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Logging/EventIds.cs (hunk 13 above)
Comment (Siddartha-tech) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Logging/EventIds.cs]: Yes updated EventId as per suggested, Thanks!

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Logging/EventIds.cs (hunk 14 above)
Comment (Siddartha-tech) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Logging/EventIds.cs:511]: Yes updated EventId as per suggested, Thanks!

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Logging/EventIds.cs (hunk 12 above)
Comment (Siddartha-tech) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Logging/EventIds.cs]: Yes updated EventId as per suggested, Thanks!

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs (hunk 15 above)
Comment (Siddartha-tech) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs]: Yes updated message as per suggested, Thanks!

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs (hunk 16 above)
Comment (Siddartha-tech) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs]: Yes updated message as per suggested, Thanks!

**Review by vijendr13545 — APPROVED** [review]
LGTM. Thanks.

**Review by martyn-fewtrell — APPROVED** [review]
Looks good

**Pull request merged**